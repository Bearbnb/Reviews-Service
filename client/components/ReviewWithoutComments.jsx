import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import styles from '../../styles/Review.css';

const moment = require('moment');

const ReviewWithoutComments = ({ review, FlagClick }) => (
  <div className={styles.reviewContainer}>
    <div>
      <div className={styles.mainHeader}>
        <div className={styles.imageHeader}>
          <img className={styles.avatar} src={review.photo} alt="" />
        </div>
        <div className={styles.textHeader}>
          <div className={styles.reviewer}>
            {review.name}
            <div className={styles.flag}>
              <button type="button" onClick={() => FlagClick()}><i className="far fa-flag" /></button>
            </div>
          </div>
          <div className={styles.date}>
            {moment(review.created).format('MMMM YYYY')}
          </div>
        </div>
      </div>
    </div>
    <div className={styles.reviewText}>
      <p>{review.review}</p>
    </div>
  </div>
);

ReviewWithoutComments.propTypes = {
  review: PropTypes.shape({
    host_comments: PropTypes.string.isRequired,
    created: PropTypes.string.isRequired,
    photo: PropTypes.string.isRequired,
  }).isRequired,
  FlagClick: PropTypes.func.isRequired,
};

export default ReviewWithoutComments;
