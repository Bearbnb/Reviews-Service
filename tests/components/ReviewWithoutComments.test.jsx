import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import ReviewWithoutComments from '../../client/components/ReviewWithoutComments';

describe('<ReviewWithoutComments />', () => {
  it('should render The summary of all reviews to the DOM', () => {
    const baseProps = {
      review: {
        host_comments: '',
        created: '',
        photo: '',
      },
      FlagClick: jest.fn(),
    };
    const wrapper = shallow(<ReviewWithoutComments {...baseProps}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render a modal when flag icon is clicked', () => {
    const baseProps = {
      review: {
        host_comments: '',
        created: '',
        photo: '',
      },
      FlagClick: jest.fn(),
    };
    const wrapper = shallow(<ReviewWithoutComments {...baseProps} />);
    wrapper.find('button').first().simulate('click');
    expect(baseProps.FlagClick).toHaveBeenCalledTimes(1);
  });
});
