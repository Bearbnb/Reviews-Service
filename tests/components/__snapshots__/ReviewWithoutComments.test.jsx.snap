// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewWithoutComments /> should render The summary of all reviews to the DOM 1`] = `
<div
  className="reviewContainer"
>
  <div>
    <div
      className="mainHeader"
    >
      <div
        className="imageHeader"
      >
        <img
          alt=""
          className="avatar"
          src=""
        />
      </div>
      <div
        className="textHeader"
      >
        <div
          className="reviewer"
        >
          <div
            className="flag"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              <i
                className="far fa-flag"
              />
            </button>
          </div>
        </div>
        <div
          className="date"
        >
          Invalid date
        </div>
      </div>
    </div>
  </div>
  <div
    className="reviewText"
  >
    <p />
  </div>
</div>
`;
